infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/actions/RestoreRedundancyMergeSegments.java

```diff
@@ -64,7 +60,9 @@ public void impl(@Nonnull CorfuRuntime runtime) throws OutrankedException,
             // Currently the state is transferred for the complete segment.
             // TODO: Add stripe specific transfer granularity for optimization.
             // Transfer the replicated segment to the difference set calculated above.
-            StateTransfer.transfer(layout, lowRedundancyServers, runtime, layout.getFirstSegment());
+            for (String lowRedundancyServer : lowRedundancyServers) {
```

Comment:
This whole merge logic assumes no more than two segments exists. It inspects the first two segments, do transfer, but merge the last two segment, which is wrong if there are multiple segments (could happen). This wrong logic has been there for a while, can we fix it this time?

Reply:
That's not true. That would have been a serious data loss issue.
We merge only the segment which we transfer.
Let's discuss this offline if I am not understanding this right.